manager.go

```diff
@@ -182,38 +184,73 @@ func listenFdFromEnv(ident string) int {
 func (m *manager) handleSignals() {
 	c := make(chan os.Signal, 1)
 	signal.Notify(c, syscall.SIGHUP)
-	_ = <-c
 
+	for _ = range c {
```

Comment:
You _could_ remove the anonymous variable here (see [example](http://play.golang.org/p/AGLJxEIf4T)), which works in later versions of Go, but you'd break compatibility with older Go versions.